config/config.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/knadh/koanf/parsers/json"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/structs"
+	"github.com/knadh/koanf/v2"
+)
+
+var ENV_FILE_NAME = "env.json"
+
+type Config struct {
+	OpenAIAPIKey   string        `koanf:"OPENAI_API_KEY"`
+	ContextTimeout time.Duration `koanf:"CONTEXT_TIMEOUT"`
+	// IgnoreFiles is a comma-separated list of file names to ignore.
+	IgnoreFiles string `koanf:"IGNORE_FILES"`
+}
+
+// DefaultConfig is the default configuration for the application.
+var defaultConfig = Config{
+	OpenAIAPIKey:   "",
+	ContextTimeout: 5 * time.Second,
+	IgnoreFiles:    "go.mod,go.sum",
+}
+
+// LoadConfig loads the configuration from the given path.
+// If the path is empty, the default configuration is used.
+func LoadConfig(k *koanf.Koanf, filePath string) (*Config, error) {
+	if err := loadDefaultConfig(k); err != nil {
+		return nil, err
+	}
+
+	if filePath != "" && fileExists(filePath) {
+		// loadFileConfig currently supports JSON files.
+		if err := loadFileConfig(k, filePath); err != nil {
+			return nil, err
+		}
+	}
+
+	config := &Config{}
+	if err := k.Unmarshal("", config); err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+// loadDefaultConfig loads the default configuration
+// into the given koanf instance.
+func loadDefaultConfig(k *koanf.Koanf) error {
+	return k.Load(structs.Provider(defaultConfig, "koanf"), nil)
+}
+
+// loadFileConfig loads the configuration from the given file
+// it currently supports JSON files.
+func loadFileConfig(k *koanf.Koanf, filePath string) error {
+	fileExt := filepath.Ext(filePath)
+
+	if strings.Contains(fileExt, "json") {
```

Comment:
Would this inadvertantly match something like `config.json.backup`

Maybe something like: `fileExt == ".json"`

Reply:
It would! Good catch, I may move away from this `filepath.Ext` call and just check string contains .json.